src/QmlControls/CommonTopMenu.qml

```diff
@@ -0,0 +1,249 @@
+﻿import QtQuick          2.12
+import QtQuick.Controls 2.4
+import QtQuick.Layouts  1.11
+import QtQuick.Dialogs  1.3
+
+import QGroundControl                       1.0
+import QGroundControl.Controls              1.0
+import QGroundControl.Palette               1.0
+import QGroundControl.MultiVehicleManager   1.0
+import QGroundControl.ScreenTools           1.0
+import QGroundControl.Controllers           1.0
+import QtGraphicalEffects                   1.12
+
+Rectangle {
+    id:     root
+    color: Qt.rgba(0, 0, 0, 0.4)
+    z : 3
+    property int currentToolbar: flyViewToolbar
+
+    readonly property int flyViewToolbar:   0
+    readonly property int planViewToolbar:  1
+    readonly property int simpleToolbar:    2
+
+    property var    _activeVehicle:     QGroundControl.multiVehicleManager.activeVehicle
+    property bool   _communicationLost: _activeVehicle ? _activeVehicle.vehicleLinkManager.communicationLost : false
+    property color  _mainStatusBGColor: qgcPal.brandingPurple
+
+    property var rootWidth: null;
+    property var rootHeight: null;
+
+    Row {
```

Comment:
The use of anchors can sometimes be replaced with more efficient layout elements. Consider to use  ColumnLayout or RowLayout.